packages/blockchain-link-utils/src/ripple.ts

```diff
@@ -1,62 +1,74 @@
-import type { Transaction } from '@trezor/blockchain-link-types';
+import type { AccountTxTransaction, ServerInfoResponse } from 'xrpl';
 
-type RippleTransactionMetadata = {
-    AffectedNodes: any[];
-    DeliveredAmount?: any[];
-    TransactionIndex: number;
-    TransactionResult: string;
-    delivered_amount?: string;
-};
+import type { ServerInfo, Transaction } from '@trezor/blockchain-link-types';
 
-// export const transformServerInfo = (payload: GetServerInfoResponse) => {
-export const transformServerInfo = (payload: any) => ({
+export const transformServerInfo = (payload: ServerInfoResponse): Omit<ServerInfo, 'url'> => ({
     name: 'Ripple',
     shortcut: 'xrp',
     network: 'xrp',
     testnet: false,
-    version: payload.buildVersion,
+    version: payload.result.info.build_version,
     decimals: 6,
-    blockHeight: payload.validatedLedger.ledgerVersion,
-    blockHash: payload.validatedLedger.hash,
+    blockHeight: payload.result.info.validated_ledger?.seq ?? 0,
+    blockHash: payload.result.info.validated_ledger?.hash ?? '',
 });
 
-// https://bitcoin.stackexchange.com/questions/23061/ripple-ledger-time-format/23065#23065
+// XRPL timestamps are based on the Ripple Epoch (2000-01-01T00:00:00Z).
+// To convert to a standard Unix timestamp, add this offset.
+// https://xrpl.org/docs/references/protocol/data-types/basic-data-types#specifying-time
 const BLOCKTIME_OFFSET = 946684800;
 
+const getUnixTimestamp = (xrplTimestamp?: number): number => {
+    if (!xrplTimestamp || xrplTimestamp <= 0) {
+        return 0;
+    }
+
+    return xrplTimestamp + BLOCKTIME_OFFSET;
+};
+
 export const transformTransaction = (
-    tx: any,
-    meta: RippleTransactionMetadata | null,
+    hash: string | undefined,
+    tx_json: NonNullable<AccountTxTransaction['tx_json']>,
+    meta: AccountTxTransaction['meta'] | undefined,
     descriptor?: string,
 ): Transaction => {
-    const blockTime =
-        typeof tx.date === 'number' && tx.date > 0 ? tx.date + BLOCKTIME_OFFSET : tx.date;
-    const isTokenTransaction = tx.Amount && typeof tx.Amount !== 'string';
+    let txType: Transaction['type'] = 'unknown';
+    let addresses: string[] = [];
+    let amount: string | undefined;
+    let destinationTag: number | undefined;
 
-    let txType: Transaction['type'];
+    if (tx_json.TransactionType === 'Payment') {
+        const deliverMax = (tx_json as { DeliverMax?: string }).DeliverMax ?? undefined;
```

Comment:
I see that the type is missing in XRPL.js, maybe would be useful to add a comment